lifx-notifier 1.2.1

ShotSkydiver opened the pull request:
![TagCheck](https://bitrise-steplib-git-check.herokuapp.com/tag?pr=3049)

### What to do if the build fails?

At the moment contributors do not have access to the CI workflow triggered by StepLib PRs. In case of a failed build, we ask for your patience. Maintainers of Bitrise Steplib will sort it out for you or inform you if any further action is needed.

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [X] __I will not move an already shared step version's tag to another commit__
- [X] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [X] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [X] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [X] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

koral-- commented on steps/lifx-notifier/1.2.1/step.yml:
```diff
@@ -0,0 +1,105 @@
+  Add this step to your workflow when color change is required.
+
+  This step can be used at the start (notify build start) and at the end (notify success/failure) of your workflow.
+  If there's no need to notify about workflow start than adding step to the end of the workflow will be enough.
```
```suggestion
  If there's no need to notify about workflow start then adding step to the end of the workflow will be enough.
```
Anyway, this sentence does not add any effective information IMO.

koral-- commented on steps/lifx-notifier/1.2.1/step.yml:
```diff
@@ -0,0 +1,105 @@
+  opts:
+    description: |
+      This step uses official LIFX API,
+      which requires OAuth2 Access token (https://api.developer.lifx.com/docs/authentication).
```
Can be markdown link

koral-- commented on steps/lifx-notifier/1.2.1/step.yml:
```diff
@@ -0,0 +1,105 @@
+    description: |
+      This step uses official LIFX API,
+      which requires OAuth2 Access token (https://api.developer.lifx.com/docs/authentication).
+    is_required: true
```
This input should be marked as sensitive

zsolt-marta-bitrise replied:
+1 on this, it is a requirement for new steps. I strongly urge you to add this change to this PR (but otherwise since this particular code is not changed, I've approved the PR itself)

koral-- commented on steps/lifx-notifier/1.2.1/step.yml:
```diff
@@ -0,0 +1,105 @@
+support_url: https://github.com/ShotSkydiver/steps-lifx-notifier/issues
+published_at: 2021-05-21T16:03:58.500425-06:00
+source:
+  git: https://github.com/ShotSkydiver/steps-lifx-notifier.git
```
Script requires curl but it is not listed as a dependency.

zsolt-marta-bitrise reviewed (approved):
Thanks @koral-- for the review! The points you made are very valid, but I'm approving this because the highlighted improvements affect the existing step config, not the changes made in this particular release.

@ShotSkydiver Would you like to apply those changes now?

mateherber commented:
As no response were received in some time I'm gonna proceed with merging this PR. @ShotSkydiver please fix the review items in a future release.

ShotSkydiver commented:
I apologize for not addressing those review comments as well as not responding! I honestly had forgotten about this PR of until just now...

I really appreciate the review! I have been meaning to take the time to overhaul the code for this step and re-do it, this step was originally created by someone else who stopped maintaining it back in early 2017, so I took it over semi-recently, so a lot of this code is old and the original maintainer didn't have English as their first language.
